packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1360,3 +1360,39 @@ class KeepAlive extends ParentDataWidget<KeepAliveParentDataMixin> {
     properties.add(DiagnosticsProperty<bool>('keepAlive', keepAlive));
   }
 }
+
+/// A sliver that constrains the cross axis extent of its sliver child.
+///
+/// The [SliverConstrainedCrossAxis] takes an [extent] parameter and uses it as
+/// the cross axis extent of the [SliverConstraints] passed to the sliver child.
+/// The widget ensures that the [extent] is a value between 0 and the parent's
+/// cross axis extent.
+///
+/// This is useful when you want to apply a custom cross-axis extent constraint
+/// to a sliver child.
+class SliverConstrainedCrossAxis extends SingleChildRenderObjectWidget {
+  /// Creates a sliver that constrains the cross axis extent of its sliver child.
+  ///
+  /// The [extent] parameters must not be null.
+  const SliverConstrainedCrossAxis({
+    super.key,
+    required this.extent,
+    Widget? sliver,
+  }) : super(child: sliver);
```

Comment:
If so, there should be an assertion here.
Also, the extent should be asserted here within valid bounds.

Reply:
I put the extent assertion in performLayout since we can't check if extent < constraints.crossAxisExtent here. I hope that's fine?

Reply:
We can still assert that it is not < 0 or > infinity at this point.

Reply:
Done!